caffe2/onnx/onnxifi_manager.cc

```diff
@@ -0,0 +1,58 @@
+#include "caffe2/onnx/onnxifi_manager.h"
+
+#include <mutex>
+
+#include "caffe2/core/logging.h"
+
+namespace caffe2 {
+namespace onnx {
+
+onnxifi_library* OnnxifiManager::AddOnnxifiLibrary(
+    const std::string& name,
+    const std::string& path,
+    const std::string& suffix) {
+  auto it = onnxifi_interfaces_.find(name);
+  if (it != onnxifi_interfaces_.end()) {
+    LOG(INFO) << "Onnx interface " << name <<  " already exists";
+    return &it->second;
+  }
+
+  onnxifi_library lib;
+  const char* suffix_cstr = suffix.empty() ? nullptr : suffix.c_str();
+  auto ret = onnxifi_load(
+      ONNXIFI_LOADER_FLAG_VERSION_1_0, path.c_str(), suffix_cstr, &lib);
```

Comment:
@Maratyszcza I'm getting the following compilation error on Windows:
```
10:41:41          C:\Jenkins\workspace\caffe2-builds\py2-cuda9.0-cudnn7-windows-build\caffe2\onnx\onnxifi_manager.cc(23): error C2664: 'int onnxifi_load(uint32_t,const wchar_t *,const char *,onnxifi_library *)': cannot convert argument 2 from 'const char *' to 'const wchar_t *' [C:\Jenkins\workspace\caffe2-builds\py2-cuda9.0-cudnn7-windows-build\build\caffe2\caffe2.vcxproj]
10:41:41          C:\Jenkins\workspace\caffe2-builds\py2-cuda9.0-cudnn7-windows-build\caffe2\onnx\onnxifi_manager.cc(24): error C3536: 'ret': cannot be used before it is initialized [C:\Jenkins\workspace\caffe2-builds\py2-cuda9.0-cudnn7-windows-build\build\caffe2\caffe2.vcxproj]
```
which means I cannot use `std::string::c_str()` as it generates `const char*` instead of `const wchar*`. Is it necessary to use `wchar*` on windows?

Reply:
Windows API natively uses `wchar_t*`. I'd suggest you use `std::wstring` on Windows, or just call
```
  auto ret = onnxifi_load(
      ONNXIFI_LOADER_FLAG_VERSION_1_0, nullptr, suffix_cstr, &lib);
```